### PR TITLE
Adds support for basic auth application

### PR DESCRIPTION
**Pull request by vijetm**

### CODE REVIEW

File: examples/okta_app_basic_auth/README.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# okta_app_bookmark
```
Comment (quantumew) [inline examples/okta_app_basic_auth/README.md]: Can you update this title?

File: examples/okta_app_basic_auth/basic_updated.tf [hunk 2]
```diff
@@ -0,0 +1,22 @@
+resource "okta_user" "user" {
+  admin_roles = ["APP_ADMIN", "USER_ADMIN"]
+  first_name  = "TestAcc"
+  last_name   = "blah"
+  login       = "test-acc-replace_with_uuid@example.com"
+  email       = "test-acc-replace_with_uuid@example.com"
+}
+
+resource "okta_group" "group" {
```
Comment (quantumew) [inline examples/okta_app_basic_auth/basic_updated.tf]: Looks like this group is unused, can you pass it into the app or nix it?

File: okta/provider.go [hunk 3]
```diff
@@ -160,6 +162,7 @@ func Provider() terraform.ResourceProvider {
 			"okta_saml_idp_signing_key":      deprecateIncorrectNaming(resourceIdpSigningKey(), idpSamlKey),
 			"okta_social_idp":                deprecateIncorrectNaming(resourceIdpSocial(), idpSocial),
 			"okta_bookmark_app":              deprecateIncorrectNaming(resourceAppBookmark(), appBookmark),
+			"okta_basic_auth_app":            deprecateIncorrectNaming(resourceAppBasicAuth(), appBasicAuth),
```
Comment (quantumew) [inline okta/provider.go]: You can drop this line. Some resources were incorrectly named and thus we are still supported them with a deprecation message. All new resources can avoid this.

File: okta/resource_okta_app_basic_auth.go [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/okta/okta-sdk-golang/okta"
+	"github.com/okta/okta-sdk-golang/okta/query"
+)
+
+func resourceAppBasicAuth() *schema.Resource {
+	return &schema.Resource{
+		CustomizeDiff: func(d *schema.ResourceDiff, v interface{}) error {
+			return nil
+		},
```
Comment (quantumew) [inline okta/resource_okta_app_basic_auth.go]: You can drop this function, looks like it is noop.

**Comment by vijetmahabaleshwar-okta** [comment]
@quantumew - I have addressed your comments and updated the website docs. Not sure if these are the only files that needs to be updated. Let me know if you need any more changes.

**Comment by vijetm** [comment]
@quantumew - I have addressed your comments and updated the website docs. Not sure if these are the only docs/files that needs to be updated. Let me know if you need any more changes.

File: website/docs/r/app_basic_auth.html.markdown [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+## Attributes Reference
+
+* `id` - ID of the Application.
+
+* `label` - The Application's display name.
+
+* `url` - The URL of the sign-in page for basic auth app.
+
+* `url` - The URL of the authenticating site for basic auth app.
```
Comment (quantumew) [inline website/docs/r/app_basic_auth.html.markdown]: I think this should be `auth_url`. Awesome work on these docs!

**Comment by quantumew** [comment]
Screen Shot 2019-11-12 at 8 51 54 AM

**Comment by varunmukka-okta** [comment]
👏 @vijetm

**Pull request merged**